File: app/src/main/java/com/example/polyquicktrade/ui/ask/AskViewModel.java
package com.example.polyquicktrade.ui.ask;

import android.util.Log;

import com.example.polyquicktrade.pojo.Enquiry;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;
import java.util.Objects;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class AskViewModel extends ViewModel {

    private MutableLiveData<ArrayList<Enquiry>> enquiriesMutableLiveData;

    public MutableLiveData<ArrayList<Enquiry>> getEnquiriesMutableLiveData() {
        if (enquiriesMutableLiveData ==null)
            enquiriesMutableLiveData = new MutableLiveData<>();
        return enquiriesMutableLiveData;
    }

    public AskViewModel() {


    }

    public void getEnquiriesFromFirebase(FirebaseFirestore db){


        final ArrayList<Enquiry> enquiries = new ArrayList<>();

        db.collectionGroup("enquiry_documents").get()
                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                    @Override
                    public void onComplete(@NonNull Task<QuerySnapshot> task) {

                        if (task.isSuccessful()){

                            for (QueryDocumentSnapshot document : Objects.requireNonNull(task.getResult())) {
                                Log.d("MY_TAG", document.getId() + " => " + document.getData());
                                enquiries.add(document.toObject(Enquiry.class));
                                if (enquiriesMutableLiveData ==null)
                                    enquiriesMutableLiveData = new MutableLiveData<>();

                                enquiriesMutableLiveData.setValue(enquiries);

                            }

                        }
                        else {
                            Log.d("MY_TAG", "AskViewModel task not successful: ");

                        }
                    }
                }).addOnFailureListener(new OnFailureListener() {
            @Override
            public void onFailure(@NonNull Exception e) {

                Log.d("MY_TAG", "Message: "+e.getMessage());
                Log.d("MY_TAG", "eString: "+ e.toString());

            }
        });
    }


}

File: app/src/main/java/com/example/polyquicktrade/ui/ask/AskViewModel.kt
package com.example.polyquicktrade.ui.ask

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.polyquicktrade.pojo.Enquiry
import com.google.firebase.firestore.FirebaseFirestore
import java.util.*

class AskViewModel : ViewModel() {


    var enquiriesLiveData: MutableLiveData<ArrayList<Enquiry>>? = MutableLiveData()

    init {
        enquiriesLiveData?.value = arrayListOf()
//        getEnquiriesFromFirebase()
    }

    fun getEnquiriesMutableLiveData(): MutableLiveData<ArrayList<Enquiry>> {
        if (enquiriesLiveData == null) enquiriesLiveData = MutableLiveData()
        return enquiriesLiveData as MutableLiveData<ArrayList<Enquiry>>
    }

    fun getEnquiriesFromFirebase(db: FirebaseFirestore) {
        val enquiries = ArrayList<Enquiry>()
        db.collectionGroup("enquiry_documents").get()
            .addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    for (document in Objects.requireNonNull(task.result)) {
                        Log.d("MY_TAG", document.id + " => " + document.data)
                        enquiries.add(document.toObject(Enquiry::class.java))
                        if (enquiriesLiveData == null) enquiriesLiveData =
                            MutableLiveData()
                        enquiriesLiveData!!.value = enquiries
                    }
                } else {
                    Log.d("MY_TAG", "AskViewModel task not successful: ")
                }
            }.addOnFailureListener { e ->
                Log.d("MY_TAG", "Message: " + e.message)
                Log.d("MY_TAG", "eString: $e")
            }
    }
}